src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -404,6 +404,51 @@ def mask_data(
         self.unset_connector_state()
         return rows_updated
 
+    def run_consent_request(
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        identity_data: Dict[str, Any],
+    ) -> bool:
+        """Execute a consent request. Return whether the consent request to the third party succeeded.
+
+        Consent preferences can be retrieved from privacy_request.consent_preferences.
+        Whether or not the customer wants the given consent preference to be executable can be retrieved
+        from consent rules attached to the policy.
+
+        Return True if 200 OK
+        """
+        logger.info(
+            "Mocking consent request - actual logic should go here for saas connectors",
+            node.address.value,
+        )
+
+        consent_rule: Optional[Rule] = policy.get_consent_rule()
+        if not consent_rule:
+            raise
+        executable_uses: List[RuleUse] = [
+            use for use in consent_rule.uses if use.executable  # type: ignore[attr-defined]
+        ]
+        consent_preferences: List[Consent] = [
+            Consent(**pref) for pref in privacy_request.consent_preferences or []
+        ]
+
+        for preference in consent_preferences:
+            if preference.data_use not in [use.key for use in executable_uses]:
+                logger.info(
+                    "Skipping updating preference: {} as it is not executable",
+                    preference.data_use,
+                )
+
```

Comment:
This sort of logic is probably better moved up on one level to `GraphTask.consent_request` as it is applicable for all connectors.  This filters the consent preferences to just the executable ones.